### ISSUE TITLE
az cli 2.11.1 completely broken: cannot import name '__version__'

### ISSUE THREAD

**Issue by benjamin2605** [Azure/azure-cli#14963]
Today, all of our DevOps pipelines requiring azure cli suddenly failed with the error

```
from azure.cli.telemetry import __version__ as azure_cli_telemetry_version
ImportError: cannot import name '__version__'
```


Installing the newest update locally showed me the same error, running az --version gives me

```
The command failed with an unexpected error. Here is the traceback:                                                                                                                                                
                                                                                                                                                                                                                   
cannot import name '__version__'                                                                                                                                                                                   
Traceback (most recent call last):                                                                                                                                                                                 
  File "/lhome/x/.local/lib/python3.6/site-packages/knack/cli.py", line 207, in invoke                                                                                                                       
    self.show_version()                                                                                                                                                                                            
  File "/lhome/x/.local/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 106, in show_version                                                                                                   
    ver_string, updates_available = get_az_version_string()                                                                                                                                                        
  File "/lhome/x/.local/lib/python3.6/site-packages/azure/cli/core/util.py", line 255, in get_az_version_string                                                                                              
    versions = _get_local_versions()                                                                                                                                                                               
  File "/lhome/x/.local/lib/python3.6/site-packages/azure/cli/core/util.py", line 242, in _get_local_versions                                                                                                
    for dist in get_installed_cli_distributions():                                                                                                                                                                 
  File "/lhome/x/.local/lib/python3.6/site-packages/azure/cli/core/util.py", line 137, in get_installed_cli_distributions
    from azure.cli.telemetry import __version__ as azure_cli_telemetry_version
ImportError: cannot import name '__version__'
```

Reverting to azure-cli 2.11.0 fixes the error

**Comment by djpugh** [comment]
Happened to us as well - how did this get released with this issue?

**Comment by yonzhan** [comment]
@fengzhou-msft  could you please help with this?

**Comment by fengzhou-msft** [comment]
@benjamin2605 @djpugh how did you install azure-cli 2.11.1? You may have an older version of telemetry in your environment.
Please update to `azure-cli-telemetry` 1.0.6 with `pip install --upgrade azure-cli-telemetry`

**Comment by djpugh** [comment]
It was on an azure pipeline task this morning - pip install azure-cli

**Comment by yungezz** [comment]
https://github.com/Azure/azure-cli/issues/14965

**Comment by djpugh** [comment]
Even running with azure-cli-telemetry in a brand new virtualenv it is failing still:

Installed packages are azure-cli-2.11.1 azure-cli-core-2.11.1 azure-cli-telemetry-1.0.6

**Comment by djpugh** [comment]
Reverting back to 2.11.0 is still installing azure-cli-telemetry-1.0.6 and failing now as well

**Comment by djpugh** [comment]
Ok please ignore the above I think this is an issue now with https://github.com/Microsoft/azure-pipelines-tasks - AzureCLITask I think

**Comment by djpugh** [comment]
/the (private) build agent having the 2.11.1 version, but an older telemetry version before that fix was performed.

**Comment by yungezz** [comment]
we're reaching out to Azure pipeline team on this also.

**Comment by benjamin2605** [comment]
> @benjamin2605 @djpugh how did you install azure-cli 2.11.1? You may have an older version of telemetry in your environment.
> Please update to `azure-cli-telemetry` 1.0.6 with `pip install --upgrade azure-cli-telemetry`
> 
> Or if you're using a clean environment, please rerun the pipeline.

Ok, yes updating the telemetry package does fix the problem locally, will test it on the pipelines next.

However, I don't understand why updating azure-cli does not also update azure-cli-telemetry, what is the reasoning behind that?

**Comment by yungezz** [comment]
hi @benjamin2605 are you using the official Azure CLI task in Azure pipeline ? Based on our sync with Azure pipeline team, Azure CLI is installed by package manager, not pip. We'd like to learn more on the install story to help understand where's the problem. thanks

**Comment by linuxunderground** [comment]
> 
> 
> > @benjamin2605 @djpugh how did you install azure-cli 2.11.1? You may have an older version of telemetry in your environment.
> > Please update to `azure-cli-telemetry` 1.0.6 with `pip install --upgrade azure-cli-telemetry`
> > Or if you're using a clean environment, please rerun the pipeline.
> 
> Ok, yes updating the telemetry package does fix the problem locally, will test it on the pipelines next.
> 
> However, I don't understand why updating azure-cli does not also update azure-cli-telemetry, what is the reasoning behind that?

Hi,
By th way, can you please add azure-cli-telemetry-1.0.6.tar.gz file at [https://pypi.org/project/azure-cli-telemetry/#files](https://pypi.org/project/azure-cli-telemetry/#files). Only .whl file is released.
Thanks

**Comment by flobeier** [comment]
Today I got the same error while trying to use `terraform plan`. Reverting to azure-cli 2.11.0 fixed the issue for the time being.

**Comment by fengzhou-msft** [comment]
@linuxunderground azure-cli-telemetry-1.0.6.tar.gz is uploaded.

**Comment by linuxunderground** [comment]
Yeah,

With azure-cli-telemetry-1.0.6, I have now a working azure-cli 2.11.1
(Gentoo Portage installation)

Thanks
--
Vincent

**Comment by Phydeauxman** [comment]
This is also an issue when setting the environment variable ARM_CLOUD_METADATA_URL to the new discovery endpoint in air gapped clouds to get AZ CLI to use the discovered environment versus the explicit environment definition.

**Comment by flobeier** [comment]
Upgrading to azure-cli-telemetry-1.0.6 fixed my issue too, azure-cli 2.11.1 now works fine for me.

**Comment by skingd** [comment]
I can confirm as well. Had the issue after updating CLI, **pip install --upgrade azure-cli-telemetry** resolves it.

**Comment by fengzhou-msft** [comment]
azure-cli 2.12.0 released and pinned to azure-cli-telemetry 1.0.6 to fix the issue.
